### DISCUSSION TITLE
MediaElement - Ability to play Audio/Video files directly from a Google Drive using File Id - Net Maui

### DISCUSSION THREAD

**Discussion by Riffy135** [CommunityToolkit/Maui#1642] [Q&A]
Hi

is it possible for MediaElement  to play Audio/Video files directly from a Google Drive using it's File Id by populating the Source field using the full path.

I am currently having to download the file(s), which is not ideal as the files can be quite large in size.

Thanks

**Comment by GeorgeLeithead** [comment]
To allow third-party apps (which your .net MAUI app will be) access to Drive files requires that the user themselves give permission to the app.  I do not think that this would be possible in the toolkit.  See [Allow third-party apps for Drive files](https://support.google.com/a/answer/6105699) for more information.  You would also need to integrate the [Google Drive API](https://developers.google.com/drive/).
